### DISCUSSION TITLE
Can Ocelot be configured to route by inbound port?

### DISCUSSION THREAD

**Discussion by danbo0001** [ThreeMammals/Ocelot#2041] [Q&A]
Apologies, I wasn't sure where I should post Ocelot questions so I put this on stackoverflow (https://stackoverflow.com/questions/78304585/can-the-ocelot-net-api-gateway-be-configured-to-route-by-inbound-port) yesterday, then I saw this discussion group and put my other question here which you guys replied to super-quick so I have reposted it here :)

The scenario is that the Ocelot .NET host listens on two ports and forwards to one of two ports downstream, depending on what port the traffic arrived on at the Ocelot host.

I have looked through the Ocelot docs (https://ocelot.readthedocs.io/en/latest/features/configuration.html) and it seems that it can only route by path, method and host header, but I am wondering if this is something that can be achieved by a combination of Ocelot and .NET (e.g. kestrel) configuration.

The actual scenario I am trying to configure for is forwarding traffic which could be either http/1.1 or http/2 to a downstream host that has to listen on a port per http protocol due to the traffic itself being unencrypted (SSL having been terminated further upstream by a load balancer).

**Comment by raman-m** [comment]
Next answer will be tomorrow! 😉 The quota for questions per day has been exceeded.

@ggnaegi @RaynaldM FYI
